File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/2023s/IZ08IO3/BridgePattern/src/BridgeShapeList.java

import java.util.List;

public abstract class BridgeShapeList
{
    public abstract void initialiseProcess();
    public abstract void proceed(ShapeBase shape);
    public abstract void finaliseProcess();
    public final void proceed(List<ShapeBase> shapes)
    {
        initialiseProcess();
        for(ShapeBase shape:shapes) proceed(shape);
        finaliseProcess();
    }
}

File: semestr-8/Jezyk C#/alexander/PCS/2023s/ID06TC1/BridgePattern/BridgePattern/BridgeShapeList.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BridgePattern
{
    public abstract class BridgeShapeList
    {
        protected virtual void InitialiseProcess() {}
        protected virtual void Proceed(ShapeBase shape) {}
        protected virtual void FinaliseProcess() {}
        public virtual void Proceed(List<ShapeBase> shapes)
        {
            InitialiseProcess();
            foreach(ShapeBase shape in shapes) Proceed(shape);
            FinaliseProcess();
        }
    }
}
